federationexample/README.md

```diff
@@ -0,0 +1,135 @@
+```
+{
+  device {                  #devicegqlserver
+    id                      #devicegqlserver
+    location {              #devicegqlserver
+      latitude              #devicegqlserver
+      longitude             #devicegqlserver
+      street                #devicegqlserver
+    }
+    camera{                 #safetygqlserver
+      id                    #safetygqlserver
+      isOn                  #safetygqlserver
+      isMultiCam            #safetygqlserver
+      cameraLocation {      #safetygqlserver
+      	latitude            #safetygqlserver
+      	longitude           #safetygqlserver
+        street              #devicegqlserver
+   	  }
+    }
+  }
+}
+```
+
+
+When the gateway parses this query, it first makes a request to devicegqlserver. It adds all the fields that exist on device server, and when it sees a field it doesnt know how to resolve it replalces it with the field func `__federation` and adds theh deviceKeys are fields nested on `__federation`.
+```
+{
+  device {                  #devicegqlserver
+    id                      #devicegqlserver
+    location {              #devicegqlserver
+      latitude              #devicegqlserver
+      longitude             #devicegqlserver
+      street                #devicegqlserver
+    }
+    _federation {
+        id
+        orgId
+    }
+  }
+}
+
+```
+
+Once it gets the repsonse, it makes another request to safetygqlserver to resolve the camera field. The args `deviceKeys` are a list of he `DeviceKey` struct with the id and org Id populated from theh results of the previous query. The field `Device-safetygqserver` runs the function passed into `FetchObjectFromKeys` in the object registration, telling it how to get the full device from the keys. 
+
+SafetyGqlserver can resolve all the fields but street name, so it once again replaces `street` with the `__federation` field func and adds the keys latitude and longtitudde as fields. 
+
+```
+{
+    _federation {
+        Device_safetygqlserver(args: deviceKeys) {
+            camera{                    #safetygqlserver
+                id                     #safetygqlserver
+                isOn                   #safetygqlserver
+                isMultiCam             #safetygqlserver
+                cameraLocation {       #safetygqlserver
+                    latitude           #safetygqlserver
+                    longitude          #safetygqlserver
+                    _federation {
+                        latitude
+                        longitude
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Once the gqlgateway gets the result from the query to safetygqlserver, it makes one more request to deviceservice to resolve street. It sends the locationKeys as arguments and resolve the field street. 
+```
+{
+    _federation {
```

Comment:
```suggestion
    __federation {
```

Reply:
We had to change it to one underscore to match the apollo schema validation. The double underscore is reserved for fields used for introspection